test/cls.test.js

```diff
@@ -0,0 +1,145 @@
+"use strict";
+
+/* jshint camelcase: false */
+var chai      = require('chai')
+  , sinon     = require('sinon')
+  , expect    = chai.expect
+  , Support   = require(__dirname + '/support')
+  , Sequelize = Support.Sequelize
+  , Promise   = Sequelize.Promise
+  , cls       = require('continuation-local-storage')
+  , current = Support.sequelize;
+
+chai.config.includeStack = true;
+
+describe(Support.getTestDialectTeaser("Continuation local storage"), function () {
+  before(function () {
+    this.sequelize = Support.createSequelizeInstance({
+      namespace: cls.createNamespace('sequelize')
+    });
+  });
+
+  beforeEach(function () {
+    return Support.prepareTransactionTest(this.sequelize).bind(this).then(function (sequelize) {
+      this.sequelize = sequelize;
+
+      this.ns = cls.getNamespace('sequelize');
+
+      this.User = this.sequelize.define('user', {
+        name: Sequelize.STRING
+      });
+      return this.sequelize.sync({ force: true });
+    });
+  });
+
+  var autoCallback = function autoCallback(sequelize, cb) {
+    return sequelize.transaction(cb);
+  };
+  var thenCallback = function thenCallback(sequelize, cb) {
+    return sequelize.transaction().then(function (t) {
+      cb().then(function () {
+        t.commit();
+      });
+    });
+  };
+
+  if (current.dialect.supports.transactions) {
+    [autoCallback, thenCallback].forEach(function (cb) {
```

Comment:
I say kill `thenCallback` support. Keep the magic to auto and make the test easier to comprehend :p